bugshaker/src/main/java/com/github/stkent/bugshaker/ActivityReferenceManager.java

```diff
@@ -26,14 +26,35 @@
 public final class ActivityReferenceManager {
 
     @Nullable
-    private WeakReference<Activity> wActivity;
+    private WeakReference<Activity> wTopActivity;
 
-    public void setActivity(@NonNull final Activity activity) {
-        this.wActivity = new WeakReference<>(activity);
+    @Nullable
+    private WeakReference<Activity> wPresentingActivity;
+
+    public void setTopActivity(@NonNull final Activity activity) {
+        wTopActivity = new WeakReference<>(activity);
+    }
+
+    public void setPresentingActivity(@NonNull final Activity activity) {
+        wPresentingActivity = new WeakReference<>(activity);
```

Comment:
I'm not understanding the need for two here, let's chat when you're out of standup